ui/src/app/guards/registration-guard.service.ts

```diff
@@ -5,29 +5,32 @@ import {
   CanActivateChild,
   Router, RouterStateSnapshot
 } from '@angular/router';
-import {Observable} from 'rxjs/Observable';
 
-import {ProfileStorageService} from 'app/services/profile-storage.service';
 import {hasRegisteredAccess} from 'app/utils/access-tiers';
+import {profileStore} from 'app/utils/stores';
+import {Profile} from 'generated/fetch';
 
 @Injectable()
 export class RegistrationGuard implements CanActivate, CanActivateChild {
-  constructor(
-    private profileStorageService: ProfileStorageService,
-    private router: Router) {}
+  constructor(private router: Router) {}
 
-  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
-    return this.profileStorageService.profile$.flatMap((profile) => {
-      if (hasRegisteredAccess(profile.accessTierShortNames)) {
-        return Observable.from([true]);
-      } else {
-        this.router.navigate(['/']);
-        return Observable.from([false]);
-      }
-    });
+  async canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<boolean> {
+    if (!profileStore.get().profile) {
```

Comment:
Who is responsible for initializing this store now, and how do we know this isn't going to happen multiple times? Is it this guard?

Reply:
It should probably be `SignedInComponent`.

Reply:
...but `SignedInComponent` won't render until `DisabledGuard` passes, argh! Perhaps `SignInService`. kick the can another block down the road.

Reply:
I'm fine with any location that works, since this is temporary. As long as it's just one location (or the store supports a "lazy get" which works from any caller)